tests/mocks/rpc-service.ts

```diff
@@ -0,0 +1,109 @@
+import { ValidBlockData } from "./handler";
+type PromiseResult = { success: boolean; rpcUrl: string; duration: number };
+
+export class RPCService {
+  static async testRpcPerformance(
+    networkId: number,
+    latencies: Record<string, number>,
+    runtimeRpcs: string[],
+    rpcHeader: object,
+    rpcBody: string,
+    rpcTimeout: number
+  ): Promise<{ latencies: Record<string, number>; runtimeRpcs: string[] }> {
+    const successfulPromises = runtimeRpcs.map<Promise<PromiseResult>>(
+      (rpcUrl) =>
+        new Promise<PromiseResult>((resolve) => {
+          const abortController = new AbortController();
+          const startTime = performance.now();
+          const timeoutId = setTimeout(() => {
+            abortController.abort();
+            resolve({ rpcUrl, success: false, duration: 0 });
+          }, rpcTimeout);
+
+          fetch(rpcUrl, {
+            method: "POST",
+            headers: Object.assign({}, rpcHeader, { "Content-Type": "application/json" }),
+            body: rpcBody,
+            signal: abortController.signal,
+          })
+            .then(() => {
```

Comment:
This is in an async function is there a reason why you used this syntax

Reply:
Gathering all of the promises synchronously and then racing them all via `await Promise.race()` and collecting them all via `await Promise.allSettled()`

This is only here to satisfy testing as jest and axios were not playing nice